File: VB/DXTREMEMVC/Controllers/StartUpController.vb
Imports Microsoft.VisualBasic
Imports System.Collections
Imports System.Web.Mvc
Imports System.Web.Script.Serialization

Namespace DXTREMEMVC.Controllers
	Public Class StartUpController
		Inherits Controller
		Public Function StartUpIndex() As ActionResult
			Dim model As Object = CreateScriptObjectFromDataObject(GetDataObject())
			Return View(model)
		End Function

		Private Function GetDataObject() As IEnumerable
			Return New DXTREMEMVC.Models.DataHelper().GetItems()
		End Function

		Private Function CreateScriptObjectFromDataObject(ByVal dataObject As IEnumerable) As String
			Dim serializer As New JavaScriptSerializer()
			Return serializer.Serialize(dataObject)
		End Function
	End Class
End Namespace

File: CS/DXTREMEMVC/Controllers/StartUpController.cs
using System.Collections;
using System.Web.Mvc;
using System.Web.Script.Serialization;

namespace DXTREMEMVC.Controllers {
    public class StartUpController : Controller {
        public ActionResult StartUpIndex() {
            object model = CreateScriptObjectFromDataObject(GetDataObject());
            return View(model);
        }

        private IEnumerable GetDataObject() {
            return new DXTREMEMVC.Models.DataHelper().GetItems();
        }

        private string CreateScriptObjectFromDataObject(IEnumerable dataObject) {
            JavaScriptSerializer serializer = new JavaScriptSerializer();
            return serializer.Serialize(dataObject);
        }
    }
}
